apps/dirac/Ramses/ramses.py

```diff
@@ -0,0 +1,94 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+#------------------------------------------------------------------------------------------------------------------------------------
+# Base class for Ramses.
+# Also defines the sanity test.
+#------------------------------------------------------------------------------------------------------------------------------------
+class RamsesMPI(rfm.RunOnlyRegressionTest):
+    
+    def __init__(self):
+
+        self.time_limit = '0d0h10m0s'
+        self.exclusive_access=True
+        self.valid_systems = ['*']
+        self.valid_prog_environs = ['*']
+        self.executable = './ramses3d'
```

Comment:
Not for this PR, but in the base class it might be nice to have an assert that this exists before running a benchmark (unless of course ReFrame does this already)

Reply:
I would need more discussion on this one as I did not understand it completely. As you have mentioned that it might be for another PR.

Reply:
I was thinking that the following could be in a base test class (i.e. something like that in #19) 

```python
class BaseTest(rfm.RegressionTest):
    @run_after("something?")
    def check_executable_exists(self):

        if not Path(self.executable).exists():
            raise RuntimeError("Cannot run {self} - executable didn't exist")
```

to prevent a benchmark from queueing only to fail because the executable doesn't exist. Perhaps ReFrame does this already 🤔 ?

Reply:
Thanks for clarifying and yes, it is a nice suggestion. I think @giordano might be having more knowledge about this.

I am not sure that Reframe will check about the executable automatically. The test will be queued and will fail because executable is not found. If we insert the above statement, I think, it will check to see if executable exists or not. However, the test may be scheduled and only when it runs it will find about the missing executable (again not entirely sure).